### PR TITLE
Make  `BaseTestRawIO` setup per class and not per test

### PR DESCRIPTION
**Pull request by h-mayorquin**
Related to the discussion in #1550

### CODE REVIEW

**Comment by zm711** [comment]
Looks like maxwell testing relies directly on setUp? Do you want to dig into or you want me to? 

```
FAILED neo/test/rawiotest/test_maxwellrawio.py::TestMaxwellRawIO::test_read_all - AttributeError: type object 'BaseTestRawIO' has no attribute 'setUp'
```

**Comment by h-mayorquin** [comment]
Fixed.

File: neo/test/rawiotest/common_rawio_test.py [hunk 1]
```diff
@@ -52,21 +52,22 @@ class BaseTestRawIO:
     rawioclass = None  # the IOclass to be tested
 
     entities_to_test = []  # list of files to test compliances
-    entities_to_download = []  # when files are at gin
+    entities_to_download = []  # when files a   re at gin
```
Comment (zm711) [inline neo/test/rawiotest/common_rawio_test.py:55]:
```suggestion
    entities_to_download = []  # when files are at gin
```
haha itchy space bar finger?

**Comment by zm711** [comment]
@nikhilchandra, want to test to see if this fixes your issues using the testing locally?

**Comment by h-mayorquin** [comment]
I want to add that I think this is a good improvement on its own for tests with multiple cases so the download is not attempted twice anyway.

**Comment by zm711** [comment]
> I want to add that I think this is a good improvement on its own for tests with multiple cases so the download is not attempted twice anyway.

I agree. I was hoping for a magic speed-up. But it looks like we go from 7-8  min to 6-7 min. So a slight (although not statistically tested) in test speed.

**Comment by h-mayorquin** [comment]
1 minute is great : )

**Comment by nikhilchandra** [comment]
Will give it a shot, setting up a Linux box for testing.

**Comment by nikhilchandra** [comment]
@zm711 will give it a shot in a bit, thanks!

**Review by zm711 — APPROVED** [review]

**Pull request merged**